### ISSUE TITLE
Migrate to `tap@18` from `jest`

### ISSUE THREAD

**Issue by gurgunday** [fastify/fastify-awilix#132]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

**Comment by kibertoad** [comment]
@gurgunday Why nod `node:test`?

**Comment by gurgunday** [comment]
Yeah I actually prefer node:test, I just said tap because everything else uses it right now

+1

**Comment by kibertoad** [comment]
@mcollina Which minimum Node version are we planning to target for fastify 5? If it's 18+, `node:test` is likely to be very viable.

**Comment by jsumners** [comment]
It would be silly of us to baseline anything less than 18.

**Comment by puskin94** [comment]
I could give this a shot today!

**Comment by kibertoad** [comment]
@puskin94 which framework do you plan to use?

**Comment by puskin94** [comment]
@kibertoad currently having fun with node:test and node:assert !
